src/components/inputs/Selector.tsx

```diff
@@ -13,21 +13,46 @@ export enum SelectorType {
   STATUS_SELECTOR = 'statusSelector',
 }
 
+interface customOptions {
+  id: string
+  name: string
+  value?: string
+  //other custom types..
+}
```

Comment:
I think this way of extending `Selector` component to render `customOptions` will make it constraint for only one use case. Can we consider for more flexible use case? 
I think we should rely on passing `ReactNode` as `extraOptionRenderer`. If you need help with this then let's have sync up.